cityofaustin/atd-moped
# Use project list view query hooks for pagination and sort

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/13844
Gathered feedback on the POC here. ~This may get some clean up before PRing into this branch further.~

https://github.com/cityofaustin/atd-data-tech/issues/14025
This is now the PR for updating pagination and sorting with a custom hook to build the graphql query. The big change here is using React hooks to replace the set/get methods of the gqlabstract and remapping where that state is read and set.

## Testing
**URL to test:** 

https://deploy-preview-1137--atd-moped-main.netlify.app/moped/projects

**Steps to test:**
1. Go to the deploy preview above to go to the project list view
2. Test the pagination controls. If you test locally with production data, you can see all ~475 projects to see the paging work.
3. Test the sort order by clicking the column headers to alternate between asc and desc.
4. Notice that columns that have sorting disabled in the config like **Signal IDs** are not sortable in the UI
5. Don't worry about testing the search, advanced search, export, or any other table features not covered in these steps. Those will be covered in future issues.

---
#### Ship list
